packages/template-chakra-storefront/src/pages/product-list/partials/empty-results.jsx

```diff
@@ -9,18 +9,43 @@ import React, {Fragment} from 'react'
 import {Button, Text, Flex, Stack, Link} from '@chakra-ui/react'
 import PropTypes from 'prop-types'
 import {Link as RouteLink} from 'react-router-dom'
-import {defineMessage, FormattedMessage, useIntl} from 'react-intl'
 import {SearchIcon} from '../../../components/icons'
 import RecommendedProducts from '../../../components/recommended-products'
 import {EINSTEIN_RECOMMENDERS} from '../../../../config/constants'
+import {withTranslations, defineMessages} from '../../../components/with-translations'
 
-const contactUsMessage = defineMessage({
-    id: 'empty_search_results.link.contact_us',
-    defaultMessage: 'Contact Us'
+const messages = defineMessages({
+    contactUs: {
+        id: 'empty_search_results.link.contact_us',
+        defaultMessage: 'Contact Us'
+    },
+    cantFindAnythingForCategory: {
+        id: 'empty_search_results.info.cant_find_anything_for_category',
+        defaultMessage: 'We couldn’t find anything for {category}. Try searching for a product or {link}.'
+    },
+    cantFindAnythingForQuery: {
+        id: 'empty_search_results.info.cant_find_anything_for_query',
+        defaultMessage: 'We couldn’t find anything for "{searchQuery}".'
+    },
+    doubleCheckSpelling: {
+        id: 'empty_search_results.info.double_check_spelling',
+        defaultMessage: 'Double-check your spelling and try again or {link}.'
+    },
+    topSellers: {
+        id: 'empty_search_results.recommended_products.title.top_sellers',
+        defaultMessage: 'Top Sellers'
+    },
+    mostViewed: {
+        id: 'empty_search_results.recommended_products.title.most_viewed',
+        defaultMessage: 'Most Viewed'
+    },
+    test:{
+        id: 'empty_search_results.test',
+        defaultMessage: 'Test123'
+    }
 })
 
-const EmptySearchResults = ({searchQuery, category}) => {
-    const intl = useIntl()
+const EmptySearchResults = ({searchQuery, category, messages}) => {
```

Comment:
Now the `messages` are injected as a prop like this.. unfortunately, the lsp no longer knows what's inside the messages. So if I type `messages.something`, the lsp won't know how to auto complete it.